cpp/core.cc

```diff
@@ -211,48 +214,62 @@ void SignalHandler::Update(int sig_num) {
 }
 
 static List<int>* AllocSignalQueue() {
+  NO_ROOTS_FRAME(FUNC_NAME);  // NewList() does it
   List<int>* ret = NewList<int>();
   ret->reserve(kMaxSignalsInFlight);
   return ret;
 }
 
 List<int>* SignalHandler::TakeSignalQueue() {
+  NO_ROOTS_FRAME(FUNC_NAME);  // AllocSingalQueue() does it
   List<int>* new_queue = AllocSignalQueue();
   List<int>* ret = signal_queue_;
   signal_queue_ = new_queue;
   return ret;
 }
 
 void Sigaction(int sig_num, sighandler_t handler) {
+  NO_ROOTS_FRAME(FUNC_NAME);  // no allocations here
   struct sigaction act = {};
   act.sa_handler = handler;
   assert(sigaction(sig_num, &act, nullptr) == 0);
```

Comment:
While we're in here, I think we could change these to throw IOError or OSError -- it shouldn't matter which one, because we catch IOError_OSError